### DISCUSSION TITLE
How should Extensions/Integration project be consumed?

### DISCUSSION THREAD

**Discussion by danielalvsaaker** [gircore/gir.core#1137] [General]
The readme says the following about this project:

> [src/Extensions/Integration](https://github.com/gircore/gir.core/blob/main/src/Extensions/Integration): Optional source generators to reduce boilerplate code in your projects.

I cannot find any examples or descriptions on how to use these source generators in projects. I tried to import the Integration project into one of the GTK samples, without luck. Quickly skimming through the source generators, I see some mentions of a Template attribute, which was defined in the Gtk-3.0 package until it was removed.

Seeing as Extensions/Integration isn't packaged, is this simply unmaintained as of now and not intended for consumption?

**Comment by danielalvsaaker** [comment]
I found #393, so I assume this is not functional as of now.

**Reply by badcel** [reply]
Your assumptions are correct.

The project was used to provide support for GTK 3 composite templates.

The support for composite templates was removed during a rewrite not because it is unwanted but because the more important fundamentals were lacking behind in the generator. This holds true to this day. I did not expect it to take this long to restore composite widget feature.

So currently the project is of no use and therefore it is not published anymore.

Perhaps it will return for the 0.6.0 release, not for composite widgets, but to help a user with inheritance for the in the works update of the GObject internals. This heavily reduces the use of reflection and "magic" in the codebase. But this is not decided yet.
